src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -0,0 +1,205 @@
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <algorithm>
+#include <numeric>
+#include "micro_benchmark_basic_fixture.hpp"
+
+namespace hyrise {
+
+const int32_t MB = 1000000;
+
+class FileIOMicroReadBenchmarkFixture : public MicroBenchmarkBasicFixture {
+ public:
+  uint64_t control_sum = uint64_t{0};
+  std::vector<uint32_t> numbers;
+
+  void SetUp(::benchmark::State& state) override {
+    // TODO(everybody): Make setup/teardown global per file size to improve benchmark speed
+    ssize_t BUFFER_SIZE_MB = state.range(0);
+
+    // each int32_t contains four bytes
+    uint32_t vector_element_count = (BUFFER_SIZE_MB * MB) / 4;
+    numbers = std::vector<uint32_t>(vector_element_count);
+    for (size_t index = 0; index < vector_element_count; ++index) {
+      numbers[index] = std::rand() % UINT32_MAX;
+    }
+    control_sum = std::accumulate(numbers.begin(), numbers.end(), uint64_t{0});
```

Comment:
Fine for me. Fun question: How large could the vector be so that there is no overflow (for sure)?

Reply:
uint32 max value is 4294967295 -> 2^32 − 1 
uin64 max value is 18,446,744,073,709,551,615 -> 2^64 − 1 

To get to overflow, we need to multiply the entry count with the uint32 max value. When this exceeds the max uint64 value we're overflowing. Resolving for entry count we end up with:

Entry count = max uint 64 / max uint32 = (2^64 − 1) / (2^32 − 1) = 2^32 - 1

Hopefully thats correct :grin:

Reply:
That would be for having 2^32 elements which each have a value of 2^32 -1 right?